style/layer/poi.js

```diff
@@ -0,0 +1,82 @@
+import * as label from "../constants/label.js";
+
+const poiIcon = [
+  "match",
+  ["get", "class"],
+  "bar",
+  "martini-glass",
+  "beer",
+  "martini-glass",
+  "hospital",
+  "hospital-cross",
+  "dot_city",
+];
+
+const poiFilter = [
+  "any",
+  ["==", ["get", "class"], "bar"],
+  ["==", ["get", "class"], "beer"],
+  ["==", ["get", "class"], "hospital"],
+];
+
+const iconColor = [
+  "match",
+  ["get", "class"],
+  "bar",
+  "#0000dd",
+  "beer",
+  "#0000dd",
+  "hospital",
+  "#ff0000",
+  "#000000",
+];
+
+const labelPaint = {
+  "text-halo-color": "#ffffff",
+  "icon-halo-color": "#ffffff",
+  "text-halo-width": 1.5,
+  "icon-halo-width": 1.5,
+  "text-halo-blur": 1,
+  "icon-halo-blur": 1,
+  "text-color": iconColor,
+  "icon-color": iconColor,
+};
+
+export const poi = {
+  id: "poi",
+  type: "symbol",
+  paint: labelPaint,
+  filter: poiFilter,
+  layout: {
+    "text-font": ["Metropolis Bold"],
+    "text-size": {
+      base: 1.0,
+      stops: [
+        [5, 8],
+        [8, 10],
+        [12, 12],
+      ],
+    },
+    "icon-image": poiIcon,
+    "icon-size": 1,
+    "text-field": label.name_en,
+    "text-anchor": "bottom",
+    "text-variable-anchor": [
+      "bottom",
+      "bottom-right",
+      "bottom-left",
+      "right",
+      "left",
```

Comment:
Should `left` and `right` come first? Should `bottom-right` and `bottom-left` even be here?

In the screenshot above, “Boom Boom Room” demonstrates how difficult it can be to associate the label with the icon in a dense neighborhood. Since I’m unfamiliar with this neighborhood, I have to do process of elimination to determine that “Boom Boom” isn’t a bar to the northeast of “Room”. A few factors contribute to this difficulty:

* `text-max-width` is wide relative to `text-radial-offset` and the icon size.
* `text-radial-offset` is large compared to `text-height`.
* Line wrapping is enabled.
* `bottom-right` and `bottom-left` are less common for POI icons.